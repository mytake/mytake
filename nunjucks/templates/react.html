{% extends "base.html" %}

{% block endbody %}
  <!-- Dependencies -->
  <script src="https://unpkg.com/react@15/dist/react.js"></script>
  <script src="https://unpkg.com/react-dom@15/dist/react-dom.js"></script>

  <!-- Main -->
  <script src="./bundle-0.0.3.js"></script>
{% endblock %}
