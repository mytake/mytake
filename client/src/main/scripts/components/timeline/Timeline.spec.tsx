/*
 * MyTake.org website and tooling.
 * Copyright (C) 2017-2018 MyTake.org, Inc.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 *
 * You can contact us at team@mytake.org
 */
import * as React from "react";
import * as renderer from "react-test-renderer";
import { Corpus } from "../../common/social/social";
import { timelineItems } from "../../utils/testUtils";
import Timeline from "./Timeline";

// These tests do not work as expected. Vis.Timeline is a Portal component.
// The timelineItems are "rendered" outside of the root React component in the
// virtual DOM. These snapshots just show empty divs. Not sure how to test this.
// https://github.com/airbnb/enzyme/issues/252 might be a start.

test("Debates Timeline", () => {
  const tree = renderer
    .create(
      <Timeline
        timelineItems={timelineItems}
        onItemClick={jest.fn()}
        selectedOption={Corpus.Debates}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test("Documents Timeline", () => {
  const tree = renderer
    .create(
      <Timeline
        timelineItems={timelineItems}
        onItemClick={jest.fn()}
        selectedOption={Corpus.Documents}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
