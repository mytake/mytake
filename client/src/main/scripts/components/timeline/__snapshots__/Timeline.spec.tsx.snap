// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Debates Timeline 1`] = `
<div
  id="mytimeline"
/>
`;

exports[`Documents Timeline 1`] = `
<div
  id="mytimeline"
/>
`;
